### PR TITLE
Camera::HFOV override for Ogre and Ogre2

### PR DESCRIPTION
**Pull request by AzulRadio**
Signed-off-by: youhy <haoyuan2019@outlook.com>

# 🦟 Bug fix

Original Issue: #500

## Summary

`Camera::HFOV()` only returned a fixed value previously. Now it should change as we resize the window and change FOV at runtime. 

demo_fov

My VM can only run Ogre so I need someone to help me test this on Ogre2.

Draft because it fails too many tests

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: ogre/src/OgreCamera.cc [hunk 1]
```diff
@@ -66,6 +66,15 @@ void OgreCamera::Destroy()
   }
 }
 
+//////////////////////////////////////////////////
+math::Angle OgreCamera::HFOV() const
+{
+  double vfov = this->ogreCamera->getFOVy().valueRadians();
+  double hFOV;
+  hFOV = 2.0 * atan(tan(vfov / 2.0) * this->ImageWidth() / this->ImageHeight());
```
Comment (jennuine) [inline ogre/src/OgreCamera.cc:74]: Currently this causes a failure in `UNIT_Camera_TEST`:
```
  /github/workspace/src/Camera_TEST.cc:75: Failure
  Expected equality of these values:
    hfov.Radian()
      Which is: 1.5700000000000001
    camera->HFOV().Radian()
      Which is: 1.2862402200698853
```

I suggest you update this section of code (and for `Ogre2`) as so:

```suggestion
  double hfov = 2.0 * atan(tan(vfov / 2.0) * this->AspectRatio());
```

You will also need to update: https://github.com/gazebosim/gz-rendering/blob/1e415bb1a3d374b15d9c6cb86be76b2b269711fb/src/Camera_TEST.cc#L75

to use `EXPECT_NEAR`

Comment (AzulRadio) [reply ogre/src/OgreCamera.cc]: fixed in df3d48d.

File: ogre/src/OgreCamera.cc [hunk 2]
```diff
@@ -66,6 +66,14 @@ void OgreCamera::Destroy()
   }
 }
 
+//////////////////////////////////////////////////
+math::Angle OgreCamera::HFOV() const
+{
+  double vfov = this->ogreCamera->getFOVy().valueRadians();
+  double hFOV = 2.0 * atan(tan(vfov / 2.0) * this->AspectRatio());
```
Comment (jennuine) [inline ogre/src/OgreCamera.cc:73]: nit (to align with `vfov` which is used throughout):
```suggestion
  double hfov = 2.0 * atan(tan(vfov / 2.0) * this->AspectRatio());
```

Same for `Ogre2Camera::HFOV`

Comment (AzulRadio) [reply ogre/src/OgreCamera.cc:73]: Not using lowercase because there is a "member variable with the same name" compiler warning.

**Review by iche033 — APPROVED** [review]
looks good to me

**Pull request merged**